src/ripple/rpc/handlers/CrawlShards.cpp

```diff
@@ -0,0 +1,82 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2018 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/app/main/Application.h>
+#include <ripple/net/RPCErr.h>
+#include <ripple/nodestore/DatabaseShard.h>
+#include <ripple/overlay/Overlay.h>
+#include <ripple/protocol/JsonFields.h>
+#include <ripple/resource/Fees.h>
+#include <ripple/rpc/Context.h>
+
+namespace ripple {
+
+static std::uint32_t constexpr hopLimit = 3;
+
+/** RPC command that reports stored shards by nodes.
+    {
+        // Determines if the result includes node public key.
+        // optional, default is false
+        pubkey: <bool>
+
+        // The maximum number of peer hops to attempt.
+        // optional, default is zero, maximum is 3
+        limit: <integer>
+    }
+*/
+Json::Value
+doCrawlShards(RPC::Context& context)
+{
+    std::uint32_t hops {0};
+    if (auto const& jv = context.params[jss::limit])
+    {
+        if (!(jv.isUInt() || (jv.isInt() && jv.asInt() >= 0)))
```

Comment:
This is just hard to read as is. My first read-through, I thought this was a bug, and you meant to have `< 0` instead of `>= 0`. It's unfortunate that our JSON library forces us to write such ugly code, or that we don't have a good way to enforce a schema.

No change needed, but if you can do something better, that'd be awesome.